include/SDL3/SDL_filesystem.h

```diff
@@ -236,6 +236,125 @@ typedef enum
  */
 extern DECLSPEC char *SDLCALL SDL_GetUserFolder(SDL_Folder folder);
 
+
+/* Abstract filesystem interface */
+
+typedef enum SDL_StatPathType
+{
+    SDL_STATPATHTYPE_FILE, /**< a normal file */
+    SDL_STATPATHTYPE_DIRECTORY, /**< a directory */
+    SDL_STATPATHTYPE_OTHER /**< something completely different like a device node (not a symlink, those are always followed) */
+} SDL_StatPathType;
+
+typedef struct SDL_Stat
+{
+    SDL_StatPathType filetype;  /* is this a file? a dir? something else? */
+    Sint64 filesize;    /* size in bytes */
+    Sint64 modtime;     /* SDL filesystem timestamp */
+    Sint64 createtime;  /* SDL filesystem timestamp */
+    Sint64 accesstime;  /* SDL filesystem timestamp */
```

Comment:
Why not use Uint64 for these?

Reply:
Filesize can be -1 to signal it's unsupported by the implementation. But maybe that's not useful.

Timestamps can be negative, though: you can have timestamps that are before the epoch.

Reply:
Maybe -1 for filesize should be documented? Or did I miss that?

Reply:
It's not documented at the moment (and might not be a good idea at all).

Reply:
People might want to serialize datetimes.
So we might need to separate the serialization from the internal representation.
related: https://github.com/libsdl-org/SDL/issues/9138

Reply:
I think time_t is signed, but it turns out FILETIME is _not_ (MSDN recommends you assign the 32-bit chunks to a `ULARGE_INTEGER`, which is unsigned), so I'm fine with this. Filesystem timestamps aren't meant to be universal representations of date and time for every conceivable use.

Reply:
This has changed in 4ed69b84433232ceebb764f38d80f9fded91cfbe ... file sizes and timestamps are unsigned, and SDL file time is nanoseconds since the Unix epoch.